File: server/src/protocol/GuardStatusResponse.cpp
#include "GuardStatusResponse.h"

#include "ResponseVisitor.h"

using namespace std;
using namespace communication;

GuardStatusResponse::GuardStatusResponse(GuardStatus type)
        : Response(10, ResponseType::GUARD_STATUS),
          type_(type)
{}

GuardStatusResponse::~GuardStatusResponse()
{}

vector<uint8_t> GuardStatusResponse::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(responseType_));
    frame.push_back(static_cast<uint8_t>(type_));

    return frame;
}

string GuardStatusResponse::getName()
{
    return string("GuardStatusResponse");
}

void GuardStatusResponse::accept(ResponseVisitor &visitor)
{
    visitor.visit(*this);
}

GuardStatus GuardStatusResponse::getGuardStatus() const
{
    return type_;
}

void GuardStatusResponse::setGuardStatus(GuardStatus type)
{
    type = type_;
}

void GuardStatusResponse::initializeDataSize()
{
    uint16_t dataSize = sizeof(responseType_);
    dataSize += sizeof(type_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/GuardStatusResponse.java
package wpam.mobile_client.protocol;

import java.util.ArrayList;

public final class GuardStatusResponse extends Response
{
	public GuardStatusResponse(GuardStatus type)
	{
		super(10, ResponseType.GUARD_STATUS);
		this.type_ = type;
	}
	public final void close()
	{
		super.close();
	}

	@Override
	public ArrayList<Integer> getFrameBytes()
	{
		initializeDataSize();

		ArrayList<Integer> frame = getHeader();
		frame.add(responseType_.getValue());
		frame.add(type_.getValue());

		return frame;
	}
	@Override
	public String getName()
	{
		return new String("GuardStatusResponse");
	}
	@Override
	public void accept(ResponseVisitor visitor)
	{
		visitor.visit(this);
	}

	public GuardStatus getGuardStatus()
	{
		return type_;
	}
	public void setGuardStatus(GuardStatus type)
	{
		type = type_;
	}

	@Override
	protected void initializeDataSize()
	{
		setDataSize(2);
	}

	private GuardStatus type_;
}